web/client/components/map/enhancers/autoMapType.js

```diff
@@ -1,4 +1,16 @@
 import { connect } from 'react-redux';
 import { createSelector } from 'reselect';
 import { mapTypeSelector } from '../../../selectors/maptype';
-export default connect(createSelector(mapTypeSelector, mapType => ({mapType})));
+import { getMapLibraryFromVisualizationMode } from '../../../utils/MapTypeUtils';
+export default connect(
+    createSelector(
+        (state, props) => {
+            // maps inside dashboards and geostories have visualizationMode included in the map config
+            if (props?.map?.visualizationMode) {
+                return getMapLibraryFromVisualizationMode(props.map.visualizationMode);
+            }
+            return mapTypeSelector(state);
+        },
+        (mapType) => ({ mapType })
+    )
+);
```

Comment:
@offtherailz I think we could also remove the global mapTypeSelector for two reason:
- each map can have a different visualization mode
- we were using the selector mainly to check if it was leaflet or openlayers, but now we can do that with getMapLibraryFromVisualizationMode

so I propose this:

```js
import { connect } from 'react-redux';
import { createSelector } from 'reselect';
import { getMapLibraryFromVisualizationMode, VisualizationModes } from '../../../utils/MapTypeUtils';
export default connect(
    createSelector(
        (state, props) => props?.map?.visualizationMode,
        (visualizationMode) => ({
            mapType: getMapLibraryFromVisualizationMode(visualizationMode || VisualizationModes._2D)
        })
    )
);

```

Reply:
I understand the point. So I think we can remove the enhancer at all, or at least restructure it. 

- It should not use `connect` anymore (no `state` is used actually). 
- it should not use `createSelector` because we do not need to create a cache.

Evaluate on your own if it makes sense to use :
- nothing, doing it inline every time
- only a utility function for this
- an enhancer, like this: 

```js
// name is a little long, maybe `toMapType` only

/**
 * Extract the `mapType` from the `map` property of a component, and return it as a property
 * in the component enhanced.
 */
visualizationModeToMapType = (Component) => props => (<Component mapType={getMapLibraryFromVisualizationMode(props?.map?.visualizationMode || VisualizationModes._2D);} {...props} />
```